lib/parse-message.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+module.exports = exports = {};
+
+
+exports.nickCommand = function(message, socket) {
+  console.log('username: ' + socket.username);
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`you are now ${socket.username}\n`);
+
+};
+
+exports.dmCommand = function(message, array){
+ // "/dm user_30 hello how are you"
+  let toUsername = message.split(' ')[1];
+  let content = message.split(' ').slice(2).join(' ').trim();
+  array.forEach(s => {
+    if (s.username === toUsername)
+      s.write(content);
+  });
+};
+
+exports.users = function(array, socket){
+  array.forEach(user => {
+    socket.write(user.username + '\n');
```

Comment:
Get in the habit of using the new ES6 interpolation, i.e. instead of using ``+`` signs to concatenate strings, the above would be:
```
`${user.username}\n`
```